superset/views/database/views.py

```diff
@@ -156,48 +157,45 @@ def form_post(self, form: CsvToDatabaseForm) -> Response:
         ).name
 
         try:
-            utils.ensure_path_exists(config["UPLOAD_FOLDER"])
+            utils.ensure_path_exists(app.config["UPLOAD_FOLDER"])
             upload_stream_write(form.csv_file.data, uploaded_tmp_file_path)
 
             con = form.data.get("con")
             database = (
                 db.session.query(models.Database).filter_by(id=con.data.get("id")).one()
             )
 
-            # More can be found here:
-            # https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html
-            csv_to_df_kwargs = {
-                "sep": form.sep.data,
-                "header": form.header.data if form.header.data else 0,
-                "index_col": form.index_col.data,
-                "mangle_dupe_cols": form.mangle_dupe_cols.data,
-                "skipinitialspace": form.skipinitialspace.data,
-                "skiprows": form.skiprows.data,
-                "nrows": form.nrows.data,
-                "skip_blank_lines": form.skip_blank_lines.data,
-                "parse_dates": form.parse_dates.data,
-                "infer_datetime_format": form.infer_datetime_format.data,
-                "chunksize": 1000,
-            }
-            if form.null_values.data:
-                csv_to_df_kwargs["na_values"] = form.null_values.data
-                csv_to_df_kwargs["keep_default_na"] = False
-
-            # More can be found here:
-            # https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_sql.html
-            df_to_sql_kwargs = {
-                "name": csv_table.table,
-                "if_exists": form.if_exists.data,
-                "index": form.index.data,
-                "index_label": form.index_label.data,
-                "chunksize": 1000,
-            }
-            database.db_engine_spec.create_table_from_csv(
+            chunks = pd.read_csv(
                 uploaded_tmp_file_path,
-                csv_table,
+                chunksize=1000,
+                encoding="utf-8",
+                header=form.header.data if form.header.data else 0,
+                index_col=form.index_col.data,
+                infer_datetime_format=form.infer_datetime_format.data,
+                iterator=True,
+                keep_default_na=not form.null_values.data,
+                mangle_dupe_cols=form.mangle_dupe_cols.data,
+                na_values=form.null_values.data if form.null_values.data else None,
+                nrows=form.nrows.data,
+                parse_dates=form.parse_dates.data,
+                sep=form.sep.data,
+                skip_blank_lines=form.skip_blank_lines.data,
+                skipinitialspace=form.skipinitialspace.data,
+                skiprows=form.skiprows.data,
+            )
+
+            df = pd.concat(chunk for chunk in chunks)
+
+            database.db_engine_spec.df_to_sql(
```

Comment:
It seems passing `database` and `table` to `db_engine_spec.df_to_sql` would introduce circular dependencies. Could we maybe create a database level instance method instead?

```python
database.fill_table(table, df)

class SqlaDatasource:
    def fill_table(self, table, **to_sql_kwargs):
        if table.is_virtual:
            raise SupersetException(_("Cannot fill a virtual table"))
       self.db_engine_spec.fill_table(
            table.name, df, schema=table.schema, **to_sql_kwargs
        )
```

Reply:
@ktmud there's no circular dependencies and this was a similar signature to `create_table_from_csv`, i.e., passing the database (model) and table.

Reply:
There is no circular imports only because `Database` type was referred with a string `"Database"` instead of the imported class. Conceptually the dependency still seems circular. `Database` depends on `DbEngineSpec` but `DbEngineSpec` also has class methods that expects `Database` or `Table` as an input.

Not sure whether this is a big deal, but if we are only using one or two props from an input argument, it's probably better to just pass the prop values to avoid tight coupling as well as allowing the function to be used elsewhere where initializing the dependent class is not required.